plugins/osdn/plugin.go

```diff
@@ -173,8 +180,23 @@ func (plugin *OsdnNode) SetUpPod(namespace string, name string, id kubeletTypes.
 }
 
 func (plugin *OsdnNode) TearDownPod(namespace string, name string, id kubeletTypes.DockerID) error {
+	hostVethName, contMac, contIP, err := plugin.vethInspect(id)
+	if err != nil {
+		return err
+	}
+
+	containerNs, err := plugin.getContainerNS(id)
+	if err != nil {
+		return fmt.Errorf("Failed to get container network namespace: %v", err)
+	}
+	defer containerNs.Close()
+
+	if _, err := plugin.runIPAM(containerNs, "DEL", id); err != nil {
```

Comment:
it's weird that TearDownPod() calls plugin.runIPAM("DEL") but SetUpPod() doesn't call plugin.runIPAM("ADD"). (because in one case it's hidden in a helper function and in the other case it isn't...)